script.js

```diff
@@ -0,0 +1,99 @@
+var imArr = [];
+var counter = 0;
+var nIntervId;
+
+
+$(function() {
+  // attach the form submission to the search function
+  $("#search-form").on('submit', search);
+  });
+
+function search(event) {
+// Stop the form from changing the page.
+  event.preventDefault();
+  clearSearchResults();
+
+  var input = $("#query")
+  var userQuery =  input.val();
+
+
+  $.get('https://www.reddit.com/search.json', {
+    q: userQuery,
+    limit: 100
+    // nsfw:no
+  }).done(function(response) {
+    console.log(response);
+  
+
+//changing based on Nat's code
+//this filters through the children to check for the preview tag & find the url
+//then pushes those urls into an array
+response.data.children.filter(function(result) {
```

Comment:
Because we are pushing into a new array via push, it makes more sense to do .forEach instead of .filter here.
You could also do a .filter and .map combo here, but .forEach is probably the best bet.
Functionally it still works the way you have it, but it is trying to return a new array that you aren't storing anywhere.